UnityProject/Assets/Scripts/Objects/VendorTrigger.cs

```diff
@@ -55,8 +113,7 @@ private bool ServerVendorInteraction(GameObject originator, Vector3 position, st
 			return false;
 		}
 
-		int randIndex = Random.Range(0, vendorcontent.Length);
-		ItemFactory.SpawnItem(vendorcontent[randIndex], transform.position, transform.parent);
+		ItemFactory.SpawnItem(itemToVend, transform.position, transform.parent);
 
 		stock--;
```

Comment:
You probably want to decrement stockAmts[itemToVend]'s stock value here instead

Reply:
Haven't worked on the actual dispensing and stocking yet. Using what was given from VendorTrigger and working off of that.

Reply:
Vend Method isn't being called by anything yet. Once the stockitem prefab is turned into some kind of button to vend a vendItem then the vend method should be looked at and rewritten from scratch. Networking checks will need to be applied here